### PR TITLE
bevy_log: Use `os_log` on iOS

### PR DESCRIPTION
**Pull request by HackerFoo**
# Objective

On mobile devices, it's best to use the OS's native logging due to the difficulty of accessing the console. This is already done for Android.

## Solution

This PR uses Absolucy's [`tracing-oslog`](https://github.com/Absolucy/tracing-oslog) ([ZLib license](https://github.com/Absolucy/tracing-oslog/blob/main/LICENSE.md)) for iOS in order to use Apple's `os_log`.

## Changelog

- Change `bevy_log` to use Apple's `os_log` on iOS.

### CODE REVIEW

File: crates/bevy_log/src/lib.rs
Comment (IceSentry) [inline crates/bevy_log/src/lib.rs:104]: SInce it's only used in iOS should this be conditionally added to the struct?

Comment (HackerFoo) [reply crates/bevy_log/src/lib.rs:105]: Done, thanks.

**Review by alice-i-cecile — APPROVED** [review]
I'm going to merge this optimistically; @HackerFoo please open a new issue or PR if you run into problems on iOS.

The code itself is good and well explained.

bors r+

**Comment by james7132** [comment]
@HackerFoo sorry for dropping the ball on this, could you update this PR?

**Comment by alice-i-cecile** [comment]
Ping @HackerFoo: I'm leaving this in the milestone because it seems like the merge conflict should be easy.

File: crates/bevy_log/Cargo.toml
Comment (alice-i-cecile) [inline crates/bevy_log/Cargo.toml:38]:
```suggestion
bevy_internal = { path = "../bevy_internal", version = "0.11.0-dev" }
```

Comment (alice-i-cecile) [inline crates/bevy_log/Cargo.toml:38]:
```suggestion
```

Comment (alice-i-cecile) [reply crates/bevy_log/Cargo.toml:38]: I don't think this is the right fix: no equivalent line exists.

Comment (alice-i-cecile) [reply crates/bevy_log/Cargo.toml:38]: Pretty sure this should just be removed.

Comment (cart) [reply crates/bevy_log/Cargo.toml:38]: Agreed this is spurious

Comment (mockersf) [reply crates/bevy_log/Cargo.toml]: this is just a *very* old PR, that's how `bevy_log` was back in the day

**Comment by cart** [comment]
Hmm multiple new conflicting build deps (due to an older version of bindgen). Not as bad as non-build deps, but also not great.

**Comment by alice-i-cecile** [comment]
>  @Alice 🌹 Here’s a newer version. I’ll try to update the PR if I can: https://github.com/HackerFoo/bevy/commit/b97063dd4547ae6023937287c56f6f03e021446c

**Comment by alice-i-cecile** [comment]
I don't have the setup or expertise to validate this independently: adding `Adopt-Me` in the hopes that someone who does can chip in :)

**Comment by therealbnut** [comment]
@alice-i-cecile Hi, I thought this "S-Adopt-Me" PR would be a good first simple Bevy contribution. I've merged the latest `main` and made some changes to get it working. I've also tested it on an iOS device and it works as I believe it's expected to.

This is my branch: https://github.com/therealbnut/bevy/tree/use-oslog-for-ios

I believe I've preserved the PR's orignal intent after the merge conflict, but as the code is changed and simplified now it may be worth a quick look to check I haven't removed something by mistake.

It's also worth noting that the dependency this adds hasn't had bug fixes released in a few years, so we may want to consider one or more of:
 1. a feature flag to opt-in, and it would also allow `os_log` on MacOS
 2. merge as-is and have some (minor?) upstream bugs
 3. hold off on this PR until a suitable alternative dependency arises
 4. maintain our own implementation

**Future work**

In a follow-up PR it might be good to make the `subsystem` field have a better default value, like [this one](https://github.com/bevyengine/bevy/blob/main/examples/mobile/bevy_mobile_example.xcodeproj/project.pbxproj#L363). That value can be retrieved programmatically if we bind another system API, and would almost always be the correct value while the current one is unlikely to ever be correct (`Bundle.main.bundleIdentifier` in Swift, or the C/ObjC equivalent for posterity).

**Comment by therealbnut** [comment]
> can you open a PR with those comments in a Questions for Reviewers section

Thanks @alice-i-cecile, I've made that PR [here](https://github.com/bevyengine/bevy/pull/13364), I hope that's what you meant.

**Comment by richchurcher** [comment]
Closing in favour of #13364.

**Comment by therealbnut** [comment]
@richchurcher thanks for the reminder, my PR on the third-party library blocking #13364 was finally merged, so hopefully that fixes the security advisories

**Pull request closed without merging**
